editoast/src/converters/utils.rs

```diff
@@ -0,0 +1,213 @@
+        }],
+    );
+    point_groups.insert(
+        "RIGHT".into(),
+        vec![SwitchPortConnection {
+            src: "BASE".into(),
+            dst: "RIGHT".into(),
+        }],
+    );
+
+    let mut cross_groups = std::collections::HashMap::new();
+    cross_groups.insert(
+        "default".into(),
+        vec![
+            SwitchPortConnection {
+                src: "NORTH".into(),
+                dst: "SOUTH".into(),
+            },
+            SwitchPortConnection {
+                src: "EASTH".into(),
+                dst: "WEST".into(),
+            },
+        ],
+    );
+
+    let mut double_cross_groups = std::collections::HashMap::new();
+    double_cross_groups.insert(
+        "n1-s1".into(),
+        vec![SwitchPortConnection {
+            src: "NORTH-1".into(),
+            dst: "SOUTH-1".into(),
+        }],
+    );
+    double_cross_groups.insert(
+        "n2-s1".into(),
+        vec![SwitchPortConnection {
+            src: "NORTH-1".into(),
+            dst: "SOUTH-2".into(),
+        }],
+    );
+    double_cross_groups.insert(
+        "n1-s2".into(),
+        vec![SwitchPortConnection {
+            src: "NORTH-2".into(),
+            dst: "SOUTH-1".into(),
+        }],
+    );
+    double_cross_groups.insert(
+        "n2-s2".into(),
+        vec![SwitchPortConnection {
+            src: "NORTH-2".into(),
+            dst: "SOUTH-2".into(),
+        }],
+    );
+
+    vec![
+        SwitchType {
+            id: "point".into(),
+            ports: vec!["BASE".into(), "LEFT".into(), "RIGHT".into()],
+            groups: point_groups,
+        },
+        SwitchType {
+            id: "cross_ever".into(),
+            ports: vec!["NORTH".into(), "SOUTH".into(), "EAST".into(), "WEST".into()],
+            groups: cross_groups,
+        },
+        SwitchType {
+            id: "double_slip".into(),
+            ports: vec![
+                "SOUTH-1".into(),
+                "SOUTH-2".into(),
+                "NORTH-1".into(),
+                "NORTH-2".into(),
+            ],
+            groups: double_cross_groups,
+        },
+    ]
+}
+
+pub enum EdgeAngleError {
+    InvalidAngle(NodeId),
+}
+
+fn build_track_endpoint((edge, endpoint): (&Edge, Endpoint)) -> TrackEndpoint {
+    TrackEndpoint {
+        track: edge.id.as_str().into(),
+        endpoint,
+    }
+}
+
+pub fn build_track_section_link(
+    edge_a: (&Edge, Endpoint),
+    edge_b: (&Edge, Endpoint),
+) -> TrackSectionLink {
+    let id = if edge_a.1 == Endpoint::Begin {
+        edge_a.0.source
+    } else {
+        edge_a.0.target
+    };
+    let src = build_track_endpoint(edge_a);
+    let dst = build_track_endpoint(edge_b);
+    TrackSectionLink {
+        id: id.0.to_string().into(),
+        src,
+        dst,
+    }
+}
+
+// Given an edge and it’s enpoint, returns the coordinates used to compute the angle
+// It uses the nearest OpenStreetMap node, and the other as the the rails might do a loop
+// that would result in a bad angle
```

Comment:
This could be documentation instead of a simple comment. (There is the same problem in other places)

```suggestion
/// Given an edge and it’s enpoint, returns the coordinates used to compute the angle
/// It uses the nearest OpenStreetMap node, and the other as the the rails might do a loop
/// that would result in a bad angle
```